sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

```diff
@@ -104,6 +111,98 @@ public IScaleMonitor GetMonitor()
             return _scaleMonitor.Value;
         }
 
+        internal class EventProcessorStoredEventsManager : IDisposable
+        {
+            private SemaphoreSlim _storedEventsGuard;
+            private int _maxBatchSize;
+            private int _minBatchSize;
+
+            // This is internal for mocking purposes only.
+            internal ConcurrentDictionary<string, List<EventData>> StoredEvents { get; set; }
+
+            public bool HasStoredEvents
+            {
+                get
+                {
+                    try
+                    {
+                        _storedEventsGuard.Wait();
+                        return !StoredEvents.IsEmpty;
+                    }
+                    finally
+                    {
+                        _storedEventsGuard.Release();
+                    }
+                }
+            }
+
+            public EventProcessorStoredEventsManager(int maxBatchSize, int minBatchSize)
+            {
+                StoredEvents = new ConcurrentDictionary<string, List<EventData>>();
+                _storedEventsGuard = new SemaphoreSlim(1, 1);
+                _maxBatchSize = maxBatchSize;
+                _minBatchSize = minBatchSize;
+            }
+
+            public void PartitionClosing()
+            {
+                try
+                {
+                    _storedEventsGuard.Wait();
+                    StoredEvents.Clear();
```

Comment:
This would just be for a single partition.   Should we be clearing the entire set?

Reply:
Ignore me... wrong scope.